test/common/secret/secret_manager_impl_test.cc

```diff
@@ -132,6 +137,51 @@ name: "abc.com"
                             "Secret type not implemented");
 }
 
+TEST_F(SecretManagerImplTest, SdsDynamicSecretUpdateSuccess) {
+  Server::MockInstance server;
+  std::unique_ptr<SecretManager> secret_manager(new SecretManagerImpl());
+
+  NiceMock<Server::Configuration::MockTransportSocketFactoryContext> secret_context;
+
+  envoy::api::v2::core::ConfigSource config_source;
+  {
+    NiceMock<LocalInfo::MockLocalInfo> local_info;
+    NiceMock<Event::MockDispatcher> dispatcher;
+    NiceMock<Runtime::MockRandomGenerator> random;
+    Stats::IsolatedStoreImpl stats;
+    NiceMock<Upstream::MockClusterManager> cluster_manager;
+    NiceMock<Init::MockManager> init_manager;
+    EXPECT_CALL(secret_context, localInfo()).WillOnce(ReturnRef(local_info));
+    EXPECT_CALL(secret_context, dispatcher()).WillOnce(ReturnRef(dispatcher));
+    EXPECT_CALL(secret_context, random()).WillOnce(ReturnRef(random));
+    EXPECT_CALL(secret_context, stats()).WillOnce(ReturnRef(stats));
+    EXPECT_CALL(secret_context, clusterManager()).WillOnce(ReturnRef(cluster_manager));
+    EXPECT_CALL(secret_context, initManager()).WillRepeatedly(Return(&init_manager));
+
+    auto secret_provider = secret_manager->findOrCreateTlsCertificateProvider(
+        config_source, "abc.com", secret_context);
+    std::string yaml =
```

Comment:
nit: const

Reply:
changed to const std::string and replaced dynamic_pointer_cast with dynamic_cast. Thanks.